### ISSUE TITLE
can't resolve 'kubernetes.default' through kube-dns cluster ip

### ISSUE THREAD

**Issue by donbeave** [kubernetes/kubernetes#18934]
I have installed the latest version (1.1.3) on vultr. Whole cluster are works cross datacenter and cross regions.

My services list:

```
> kubectl get svc --all-namespaces
NAMESPACE     NAME              CLUSTER_IP   EXTERNAL_IP   PORT(S)         SELECTOR                             AGE
default       kubernetes        11.1.2.1     <none>        443/TCP         <none>                               23m
kube-system   kube-dns          11.1.2.10    <none>        53/UDP,53/TCP   k8s-app=kube-dns                     19m
```

If I tried to ran this command `nslookup kubernetes.default` inside `busybox` pod I will receive this error:

```
> kubectl exec -ti busybox -- nslookup kubernetes
Server:    11.1.2.10
Address 1: 11.1.2.10

nslookup: can't resolve 'kubernetes'
```

But actually, my DNS works well through pod's IP directly.

This is my description of `kube-dns` service:

```
> kubectl describe svc kube-dns --namespace=kube-system
Name:           kube-dns
Namespace:      kube-system
Labels:         k8s-app=kube-dns,kubernetes.io/cluster-service=true,kubernetes.io/name=KubeDNS
Selector:       k8s-app=kube-dns
Type:           ClusterIP
IP:         11.1.2.10
Port:           dns 53/UDP
Endpoints:      10.2.28.2:53
Port:           dns-tcp 53/TCP
Endpoints:      10.2.28.2:53
Session Affinity:   None
No events.
```

So, the following works well:

```
> kubectl exec -ti busybox -- nslookup kubernetes 10.2.28.2
Server:    10.2.28.2
Address 1: 10.2.28.2

Name:      kubernetes
Address 1: 11.1.2.1
```

I tried telnet to `11.1.2.10` and `10.2.28.2`, both can answer:

```
> kubectl exec -ti busybox -- /bin/sh
/ # telnet 11.1.2.10 53
Connection closed by foreign host
/ # telnet 10.2.28.2 53
Connection closed by foreign host
```

What's interesting, exactly the same configuration works without problems in Vagrant, but doesn't work in the cloud.

**Comment by mikedanese** [comment]
@ArtfulCoder

**Comment by skevy** [comment]
@donbeave any luck on this? fighting the exact same problem on an Azure deployment. worked fine with 1.1.2

**Comment by thockin** [comment]
What does /etc/resolv.conf look like when DNS is not working (inside a pod)

**Comment by donbeave** [comment]
@skevy I was solved it with one experimental flag `proxy-mode=iptables`, I running Kubernetes proxy in this way:
```bash
/opt/bin/kube-proxy \
        --master=${MASTER_HOST}:8080 \
        --kubeconfig=/etc/kubernetes/worker-kubeconfig.yaml \
        --proxy-mode=iptables \
        --logtostderr=true --v=4
```

**Comment by ArtfulCoder** [comment]
@skevy does this work for you ?

**Comment by skevy** [comment]
@ArtfulCoder yah I was able to get it working using iptables.

**Comment by galthaus** [comment]
I just had something similar and found that it could be timing related with regard to nslookup contacting the DNS server through the Cluster IP.  I found that switching busybox images to busybox:glibc from latest (which is uclibc) caused the nslookup to work.  I also found that editing /etc/resolv.conf to only leave the Cluster IP (removing 127.0.0.1 and node IP) caused the system to work regardless of busybox instance type.  I found that nslookup seems to take 10 seconds or so.  If I use wget to resolve names (e.g. wget http://kubernetes), the lookups work and are really fast.  I suspect that nslookup in busybox is dodgy and may not be the best thing for a consist test.
